### ISSUE TITLE
Nod red event loop delaying due to socket can node

### ISSUE THREAD

**Issue by Bhanu5302** [grodansparadis/node-red-contrib-socketcan#42]
In node-red-contrib-socketcan 1.2.9 version, whenever the data has been request from socket can in node for every 250ms, when socket can out node is added to the flow, the other part of the code execution is getting slower.
My CAN baudrate is configured at 125kbps

**Comment by grodansparadis** [comment]
OK I have not noticed this behavior. Would expect that it originate from lower layers but will check.  Thanks for reporting.

**Comment by Bhanu5302** [comment]
> OK I have not noticed this behavior. Would expect that it originate from lower layers but will check. Thanks for reporting.

Yeah, I also suspecting this issue from lower layers. Whenever I'm installing the node-red-contrib-socketcan v1.2.9 version, I'm getting errors due to no longer supported module, I think that may be the issue. If you are aware can you just confirm ?
![Uploading image.png…]()
![Uploading image (1).png…]()

**Comment by grodansparadis** [comment]
I can't view the pictures.  Do you get the warnings with the current version 1.2.14  also.  There was a mismatch in version requests of modules in previous versions which is solved.

**Comment by Bhanu5302** [comment]
> I can't view the pictures. Do you get the warnings with the current version 1.2.14 also. There was a mismatch in version requests of modules in previous versions which is solved.
![image](https://github.com/user-attachments/assets/0b730166-3e2c-4e92-91c6-d2c2123a42f3)

I didn't checked the latest version, bcz I don't have permissions to update the node red version..now i'm using node red 3.0.2, the latest socket module will install only on node red version >3.0.2 
Like what kind of issues in node-red-contrib-socketcan module v1.2.9 
Because whenever I'm exchanging data through socket CAN, the node red event loop is blocking for sometime

**Comment by grodansparadis** [comment]
Sorry for the delay here. I have been away from office.   In the old version there are requirements for old versions of submodules, among other old version of node-red (used for debug) which is not needed.  The dependencies section in package.json should have

"bcrypt": ">=5.0.1",
"minimist": ">=1.2.6",
"socketcan": ">=3.0.0",
 "util": ">=0.12.4"

There are some other changes but none that should  cause the behavior you see.

**Comment by jagathselvan** [comment]
Hi [grodansparadis](https://github.com/grodansparadis), which version of Node-Red is this library tested on and considered to be more stable, like Node-Red 3.1 or 4.

**Comment by grodansparadis** [comment]
Should work on 2.x, 3,x and 4.x as far as I know.

**Comment by jagathselvan** [comment]
The latest node-red-contrib-socketcan 1.2.15 required 3.1 or higher. For this 3.1 or 4 the lastest node-red version which tested one.

**Comment by grodansparadis** [comment]
That is still quite fair, with the older node-red versions really are getting old now.  It is more  about the dependencies than the socketcan code.   Are you in need of support for < 3.1  ?

**Comment by jagathselvan** [comment]
May be yes at the present our system using the Node-Red 3.0.2v where it can support up to "node-red-contrib-socketcan 1.2.9". Is there way to configure the lib to support the older version as well? If so, will it have any complications?

**Comment by grodansparadis** [comment]
The code is the same sine 0.8 or something. There was a small change when 1.1 came out. The version for dependencies is what has changed, and only (in this case) due to security issues.  If security is not an issue, it should alright to lower the required versions in package.json to suite your needs.
